src/sentry/uptime/endpoints/validators.py

```diff
@@ -47,6 +49,39 @@ class UptimeMonitorValidator(CamelSnakeSerializer):
         required=True, min_value=60, max_value=int(timedelta(days=1).total_seconds())
     )
     mode = serializers.IntegerField(required=False)
+    method = serializers.CharField(required=False, max_length=20)
+    headers = serializers.JSONField(required=False)
+    body = serializers.CharField(required=False)
+
+    def validate(self, attrs):
+        headers_dict = {}
+        method = "GET"
+        body = ""
+        url = ""
+        if self.instance:
+            headers_dict = self.instance.uptime_subscription.headers
+            method = self.instance.uptime_subscription.method
+            body = self.instance.uptime_subscription.body or ""
+            url = self.instance.uptime_subscription.url
+        request_line_size = len(
+            f"{attrs.get('method', method)} {attrs.get('url', url)} HTTP/1.1\r\n"
+        )
+
+        headers_dict = attrs.get("headers", headers_dict)
+        headers_size = sum(len(f"{key}: {value}\r\n") for key, value in headers_dict.items())
```

Comment:
I think to get the correct length in bytes we need to encode this as UTF-8. Otherwise we're just getting the number of code points (unique UTF-8 characters) and not the actual byte length.

```suggestion
        headers_size = sum(len(f"{key}: {value}\r\n".encode('utf-8')) for key, value in headers_dict.items())
```

Reply:
We should probably do the same for body too?